presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1779,6 +1806,7 @@ else if (context.getDriverInstanceCount().isPresent()) {
 
                 DriverFactory driverFactory = new DriverFactory(subContext.isInputDriver(), false, operatorFactories, subContext.getDriverInstanceCount());
                 context.addDriverFactory(driverFactory);
+                context.setGroupIdSymbol(subContext.getGroupIdSymbol());
```

Comment:
I think `UNION` where one branch is executed locally can produce `LocalExchange` with multiple inputs. See:
`explain select * from (select orderkey from orders) union all (values (2));` or
`explain select * from (values (1)) union all (values (2));`